console/algorithms/src/poseidon/helpers/state.rs

```diff
@@ -44,7 +44,7 @@ impl<E: Environment, const RATE: usize, const CAPACITY: usize> State<E, RATE, CA
 
 impl<E: Environment, const RATE: usize, const CAPACITY: usize> State<E, RATE, CAPACITY> {
     /// Returns an immutable iterator over the state.
-    pub fn iter(&self) -> impl Iterator<Item = &Field<E>> {
+    pub fn iter(&self) -> impl Iterator<Item = &Field<E>> + Clone {
```

Comment:
It could be possible to implement `Copy` here, instead of `Clone`

Reply:
Unfortunately the output of `chain` is not `Copy`, so we have to use `Clone` instead.